core/src/Kokkos_SYCL.hpp

```diff
@@ -166,10 +166,11 @@ namespace Impl {
 
 class SYCLSpaceInitializer : public Kokkos::Impl::ExecSpaceInitializerBase {
```

Comment:
These classes should also be marked as `final`, as that gives a slightly different guarantee than marking all the overridden `virtual` functions as `final` (which should remain, as that is an additional check that the `virtual` function signature is correct).

Reply:
What other guarantee else than "no other class would then be able to derive from it"?
In any case this can be addressed elsewhere (I am not convinced we should do anything about it in the first place)

Reply:
Is there any need for the specific initializers to be able to be derived from? I can add @nliber 's modifications pretty easily. Or is this something that should be voted on

Reply:
If we are marking things `final`, we should go ahead and mark the whole class as `final` (it's really weird otherwise).  Alternatively (and my slight preference), we can just mark those virtual functions as `override`.

FYI:  The `final` keyword can help with optimizations by compilers that do devirtualization.